test/onnx/gen_onnx.py

```diff
@@ -2207,6 +2207,843 @@ def if_else_test():
+                                                  [2, 1])
+
+    xt = np.ones((2)).astype(np.int64)
+    xt_tensor = helper.make_tensor(name='xt',
+                                   data_type=TensorProto.INT64,
+                                   dims=xt.shape,
+                                   vals=xt.flatten().astype(np.int64))
+
+    yt = np.random.randn(2, 1).astype(np.float)
+    yt_tensor = helper.make_tensor(name='yt',
+                                   data_type=TensorProto.FLOAT,
+                                   dims=yt.shape,
+                                   vals=yt.flatten().astype(np.float32))
+
+    then_add_node = onnx.helper.make_node('Add',
+                                          inputs=['x', 'xt'],
+                                          outputs=['then_out'])
+
+    else_mul_node = onnx.helper.make_node('Mul',
+                                          inputs=['y', 'yt'],
+                                          outputs=['else_out'])
+
+    then_body = onnx.helper.make_graph([then_add_node], 'then_body', [],
+                                       [then_out])
+
+    else_body = onnx.helper.make_graph([else_mul_node], 'else_body', [],
+                                       [else_out])
+
+    cond = np.array([1]).astype(np.bool)
+    cond_tensor = helper.make_tensor(name="cond",
+                                     data_type=TensorProto.BOOL,
+                                     dims=cond.shape,
+                                     vals=cond.astype(bool))
+    res = onnx.helper.make_tensor_value_info('res', TensorProto.FLOAT, [])
+
+    node = onnx.helper.make_node('If',
+                                 inputs=['cond'],
+                                 outputs=['res'],
+                                 then_branch=then_body,
+                                 else_branch=else_body)
+
+    return ([node], [x, y], [res], [cond_tensor, xt_tensor, yt_tensor])
+
+
+@onnx_test
+def if_then_else_both_empty_test():
+
+    empty_val = np.array([]).astype(np.int64)
+    empty_ts = helper.make_tensor(name='empty_tensor',
+                                  data_type=TensorProto.INT64,
+                                  dims=empty_val.shape,
+                                  vals=empty_val.flatten().astype(int))
+    shape_const = helper.make_node(
+        'Constant',
+        inputs=[],
+        outputs=['shape_const'],
+        value=empty_ts,
+    )
+
+    empty_val_y = np.array([]).astype(np.int64)
+    empty_ts_y = helper.make_tensor(name='empty_tensor_y',
+                                    data_type=TensorProto.INT64,
+                                    dims=empty_val_y.shape,
+                                    vals=empty_val_y.flatten().astype(int))
+    shape_const_y = helper.make_node(
+        'Constant_y',
+        inputs=[],
+        outputs=['shape_const_y'],
+        value=empty_ts_y,
+    )
+
+    else_out = helper.make_tensor_value_info('shape_const', TensorProto.INT64,
```

Comment:
For unit tests we should just test the empty shape by itself without if/else branches. There is no reason to test this with if/else branches either as you already have tests for scalars which will cover this.

Reply:
I wanted to keep the branches in since they're more realistic examples of what could possibly occur as I've seen in the resnext model

Reply:
This is more required than ever with the current set of changes as I'm using identity between branches for empty Then/Else cases. Will keep this in, but will be updated once I add const folding based on the conditional argument.